src/Nethermind/Nethermind.Trie/TrackingCappedArrayPool.cs

```diff
@@ -25,7 +25,7 @@ public TrackingCappedArrayPool() : this(0)
     public TrackingCappedArrayPool(int initialCapacity, ArrayPool<byte> arrayPool = null)
     {
         _rentedBuffers = new List<byte[]>(initialCapacity);
-        _arrayPool = arrayPool ?? ArrayPool<byte>.Shared;
+        _arrayPool = arrayPool;
```

Comment:
simpler to just do this here and remove other conditionals:
```suggestion
        _arrayPool = arrayPool ?? ArrayPool<byte>.Shared;
```

Reply:
Simpler codewise, but as a class field it is hidden behind a different type and the Jit con't devirtualize its access to underlying concrete array pool and inline it